File: src/tree_parity_machine/layer/HiddenLayer.java
package tree_parity_machine.layer;

import tree_parity_machine.NeuralNetException;
import tree_parity_machine.neuron.HiddenNeuron;
import tree_parity_machine.neuron.Neuron;

import java.util.Arrays;

public class HiddenLayer extends NetLayer {

    private double[] res;

    public HiddenLayer(int n, int k, int leftBound, int rightBound) {
        inputs = n;
        outputs = k;
        neurons = new Neuron[k];
        for (int i = 0; i < k; i++) {
            neurons[i] = new HiddenNeuron(n, leftBound, rightBound);
            neurons[i].init();
        }
    }

    public double[] getOutput(double[][] input) throws NeuralNetException {
        if (input.length != outputs)
            throw new NeuralNetException("Входной вектор не соответствует числу нейронов на скрытом слое");
        if (input[0].length != inputs)
            throw new NeuralNetException("Входной вектор не соответствует кол-ву весовых коэффициентов");
        double[] res = new double[outputs];
        for (int i = 0; i < outputs; i++)
            res[i] = neurons[i].getOutput(input[i]);
        this.res = res;
        return res;
    }

    @Override
    public String toString() {
        return "HiddenLayer{" +
                "res=" + Arrays.toString(res) +
                ", outputs=" + outputs +
                ", inputs=" + inputs +
                ", neurons=" + Arrays.toString(neurons) +
                '}';
    }
}

File: src/main/kotlin/com/jvvladimir/machine/tree_machine/layer/HiddenLayer.kt
package com.jvvladimir.machine.tree_machine.layer

import com.jvvladimir.machine.learning.LearningParadigm
import com.jvvladimir.machine.tree_machine.NeuralNetException
import com.jvvladimir.machine.tree_machine.neuron.HiddenNeuron

class HiddenLayer(n: Int, k: Int, leftBound: Int, rightBound: Int, paradigm: LearningParadigm) : NetLayer() {

    private lateinit var res: IntArray

    init {
        inputs = n
        outputs = k
        this.paradigm = paradigm
        neurons = Array(k) {
            HiddenNeuron(n, leftBound, rightBound, paradigm)
        }
        neurons.forEach { it.init() }
    }

    fun getOutput(input: IntArray): IntArray {
        if (input.size != inputs * outputs) throw NeuralNetException("Входной вектор не соответствует кол-ву весовых коэффициентов")
        val res = IntArray(outputs)
        for (i in 0 until outputs) res[i] = neurons[i].getOutput(input.copyOfRange(inputs * i, inputs * (i + 1)))
        this.res = res
        return res
    }

    override fun toString() =
            "HiddenLayer{ res= ${res.contentToString()} outputs= $outputs inputs= $inputs neurons= ${neurons.contentToString()} paradigm= $paradigm}"
}